File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/paddock/PaddockInformations.scala





















// Generated on 05/03/2013 10:35:08
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait PaddockInformations extends TypeDeserializer {
  val typeId = 132

  protected trait Type extends BaseType with Serializable {
    val typeId = PaddockInformations.typeId

    val maxOutdoorMount: Short
    val maxItems: Short

    def serialize(buf: Buffer) {
      buf.writeShort(maxOutdoorMount)
      buf.writeShort(maxItems)
    }

  }

  protected def create(maxOutdoorMount: Short, maxItems: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val maxOutdoorMount = buf.readShort()
    val maxItems = buf.readShort()

    create(maxOutdoorMount, maxItems)
  }

}

object PaddockInformations extends PaddockInformations {
  case class TypeImpl(maxOutdoorMount: Short, maxItems: Short) extends Type
  override def create(maxOutdoorMount: Short, maxItems: Short) = TypeImpl(maxOutdoorMount, maxItems)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(maxOutdoorMount: Short, maxItems: Short) = create(maxOutdoorMount, maxItems)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/PaddockInformations.java


// Generated on 05/08/2013 19:38:03
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PaddockInformations extends NetworkType {
    public static final short TYPE_ID = 132;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short maxOutdoorMount;
    public short maxItems;
    
    public PaddockInformations() { }
    
    public PaddockInformations(short maxOutdoorMount, short maxItems) {
        this.maxOutdoorMount = maxOutdoorMount;
        this.maxItems = maxItems;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(maxOutdoorMount);
        buf.writeShort(maxItems);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        maxOutdoorMount = buf.readShort();
        if (maxOutdoorMount < 0)
            throw new RuntimeException("Forbidden value on maxOutdoorMount = " + maxOutdoorMount + ", it doesn't respect the following condition : maxOutdoorMount < 0");
        maxItems = buf.readShort();
        if (maxItems < 0)
            throw new RuntimeException("Forbidden value on maxItems = " + maxItems + ", it doesn't respect the following condition : maxItems < 0");
    }
    
}
